src/plone/restapi/tests/test_field_serializer.py

```diff
@@ -0,0 +1,141 @@
+from plone.app.dexterity.behaviors.metadata import ICategorization
+from plone.dexterity.utils import createContentInContainer
+from plone.dexterity.utils import iterSchemata
+from plone.namedfile.file import NamedBlobFile
+from plone.namedfile.file import NamedBlobImage
+from plone.restapi.interfaces import IFieldSerializer
+from plone.restapi.serializer.dxfields import DefaultFieldSerializer
+from plone.restapi.testing import PLONE_RESTAPI_FUNCTIONAL_TESTING
+from unittest2 import TestCase
+from zope.component import getMultiAdapter
+from zope.interface.verify import verifyClass
+from zope.schema import getFields
+import json
+import os.path
+
+
+class TestDexterityFieldSerializing(TestCase):
+    layer = PLONE_RESTAPI_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        self.portal = self.layer['portal']
+        self.request = self.layer['request']
+
+    def test_boolean(self):
+        folder = createContentInContainer(self.portal, u'Folder',
+                                          exclude_from_nav=True)
+        value = self.serialize(folder, u'exclude_from_nav')
+        self.assertEquals(True, value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_textline(self):
+        folder = createContentInContainer(self.portal, u'Folder',
+                                          title=u'My Folder')
+        value = self.serialize(folder, u'title')
+        self.assertEquals(u'My Folder', value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_choice(self):
+        folder = createContentInContainer(self.portal, u'Folder',
+                                          language=u'lo')
+        value = self.serialize(folder, u'language')
+        self.assertEquals(u'lo', value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_date(self):
+        folder = createContentInContainer(
+            self.portal, u'Folder',
+            effective_date=datetime(2017, 10, 22, 19, 45, 1))
+        value = self.serialize(folder, u'effective')
+        self.assertEquals(u'2017-10-22T19:45:01', value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_richtext(self):
+        document = createContentInContainer(
+            self.portal, u'Document',
+            text=IRichText['text'].fromUnicode(u'<p>Some Text</p>'))
+        value = self.serialize(document, u'text')
+        self.assertEquals(u'<p>Some Text</p>', value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_tuple(self):
+        document = createContentInContainer(self.portal, u'Document')
+        ICategorization(document).subjects = (u'Foo', u'Bar')
+        value = self.serialize(document, u'subjects')
+        self.assertEquals([u'Foo', u'Bar'], value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_named_image(self):
+        image_asset_path = os.path.join(
+            os.path.dirname(__file__),
```

Comment:
Using [`pkg_resources. resource_filename()`](https://pythonhosted.org/setuptools/pkg_resources.html#resource-extraction) would be better here (relying on `__file__` breaks in zipped eggs and some other scenarios). [Example usage](https://github.com/4teamwork/ftw.crawler/blob/52884a1241b8ca99468620bb9c00a46c9b9fb3dd/ftw/crawler/tests/test_configuration.py#L12).